app/exposureNotificationsNativeModule.ts

```diff
@@ -125,20 +126,23 @@ export const resetExposures = async (
   debugModule.resetExposures(cb);
 };
 
-const ExposureEvents = new NativeEventEmitter(
-  NativeModules.ExposureEventEmitter,
-);
-
 type exposureEventCallback = (exsposures: Possible[]) => void;
 
 export const startListening = (
   cb: exposureEventCallback,
-): EventSubscription => {
-  return ExposureEvents.addListener(
-    'EXPOSURES_CHANGED',
-    (jsonString: string) => {
-      const exposures: Possible[] = JSON.parse(jsonString);
-      cb(exposures);
-    },
-  );
+): EventSubscription | null => {
```

Comment:
@johnschoeman this feels somewhat grotesque, but allocating `ExposureEvents` at the root scope caused an invariant error when running `GPS`. Would welcome an alternative suggestion, I'm not sure how idiomatic optionals are in `typescript`

Reply:
If you have a Maybe in your domain, then you have a Maybe in your domain. The best practice of trying to reduce and move Maybes to the edges of your system still applies, but ts doesnt have super nice of ergonomics around Maybe types so you see a lot of the `MyThing | null` syntax that you have here.

Reply:
that said i would try to pull the conditional up as much as possible in the flow.